### PR TITLE
Swap repo to use Jekyll instead of Pandoc.

### PR DESCRIPTION
**Pull request by vuw-ecs-kevin**
Lesson templates taken from the using-jekyll-gh-pages zip file that Greg
 Wilson had pointed me to in July 2015.

https://github.com/swcarpentry/lesson-example/tree/using-jekyll

Basically this adds an _config.yml to the repo and alters the MarkDown 
 in the 0?-lesson-name files to match  that in the lesson template.

I have a feeling I should be merging this into my gh-pages repo but am
still to get my head around that kind of thing, so any advice appreciated?

### CODE REVIEW

**Comment by vuw-ecs-kevin** [comment]
Bugger!
Didn't "git commit -a".
Have done now

**Comment by wking** [comment]
On Sun, May 08, 2016 at 07:24:59PM -0700, vuw-ecs-kevin wrote:

> Lesson templates taken from the using-jekyll-gh-pages zip file that
> Greg Wilson had pointed me to in July 2015.

There may be more recent work in this vein in [1,2].

**Comment by vuw-ecs-kevin** [comment]
Can't say that the new Look and Feel does anything for me, let alone
the fact that you appear to need to run a server just to to view it on your
own machine, but - hey ho!

Just out of interest: what was wrong with fairly static version: http://swcarpentry.github.io/shell-extras/

**Comment by wking** [comment]
On Sun, May 08, 2016 at 10:10:33PM -0700, vuw-ecs-kevin wrote:

> … the fact that you appear to need to run a server just to to view
> it on your own machine…

I agree about not liking the need for ‘jekyll serve’, but it should be
easy to remove that requirement [1](https://github.com/gvwilson/new-lesson-example/issues/21#issuecomment-215884006).

> Just out of interest: what was wrong with fairly static version:
> http://swcarpentry.github.io/shell-extras/

And nothing is “wrong” here, it's just that I expect the
new-lesson-example approach to Jekyll to benefit from communal SWC
maintenance.  If it does, it will be less work for any one lesson
maintainer than maintaining their own custom setup.

**Comment by gdevenyi** [comment]
If there's changes to the fundamental tooling of lesson templates, they'll happen maintainer-wide and be done all at once. Since there are ongoing discussions regarding where to go with that [1,2] , I point you towards contributing there, rather than this repo.

**Comment by vuw-ecs-kevin** [comment]
Happy to move a "discussion" around that there, however, the original pull
request was simply to make the shell-extra lesson Jekyll aware, whether it
be with the templates Geoff pointed me to last year, or the templates that
Trevor pointed me to.
You'll note that I have followed both lines of templating in my clone.
Would the closing of this request prevent the underking swap happening ?

**Comment by wking** [comment]
On Mon, May 09, 2016 at 03:29:57PM -0700, vuw-ecs-kevin wrote:

> You'll note that I have followed both lines of templating in my
> clone.  Would the closing of this request prevent the underking swap
> happening ?

This comment is not parsing for me.  Can you rephrase/unpack it?  Is
the “underlying swap” the shift from pandoc to Jekyll?  Why would with
PR's closure impact that transition?

**Comment by vuw-ecs-kevin** [comment]
> This comment is not parsing for me. Can you rephrase/unpack it?
> Is the “underlying swap” the shift from pandoc to Jekyll? Why would
> with PR's closure impact that transition?

Probably my umfamiliarity with the way that PRs are handled. Could
be my umfamiliarity with a lot of related things.

I saw "Closed" and took it to be because of the need to move the
"look and feel" discussion to another repo, which wasn't why I had
Requested  the original P.

**Comment by wking** [comment]
On Mon, May 09, 2016 at 04:01:21PM -0700, vuw-ecs-kevin wrote:

> I saw "Closed" and took it to be because of the need to move the
> "look and feel" discussion to another repo, which wasn't why I had
> Requested the original P.

This repo doesn't list maintainers in its README (see
swcarpentry/lesson-example#2), but:
—
  $ git log --format=%aN --first-parent origin/gh-pages

suggests it's @ChristinaLK.  So it's her call.

But my guess is @gdevenyi was saying [1](https://github.com/swcarpentry/shell-extras/pull/36#issuecomment-217929143) “if this lesson moves to
Jekyll, it will be via some SWC-maintained, Jekyll-based template.  If
you want to transition this lesson to Jekyll, do that by pushing the
SWC-maintained template forward”.  I think that's largely orthogonal
to any “look and feel” concerns.

**Comment by vuw-ecs-kevin** [comment]
> But my guess is @gdevenyi was saying [1] “if this lesson moves to Jekyll, it will be via some 
> SWC-maintained, Jekyll-based template.

I think the history to my small involvement this might be useful.

When I first came across the shell-extras lesson, I had a conversation, mostly
resulting from my umfamiliarity with the way the HTML was stored in with the MD
for GitHub purposes, and suggested, at that time, that the HTML, by virtue of being 
a "target" in the build process, shouldn't be stored as part of the repo.
GregW then entered the conversation and, having  read my elongated description of 
what I thought should happen, told me about already-in the-pipeline move to Jekyll, 
including the _site directory into which what I thinbk of as "make targets" would go, 
thereby cleaning up what each cloned repo would be commiting, which mirroed 
pretty much everything I though should happen. Synchronicity or what!?

I have thus assumed that everything would be moving to Jekyll/_site but hadn't
seen that in Christina's main repo, so thought to follow up.

That's the reason for my PR: all about the "substance" - nothing to do with the "style".

**Comment by wking** [comment]
On Mon, May 09, 2016 at 04:34:12PM -0700, vuw-ecs-kevin wrote:

> That's the reason for my PR: all about the "substance" - nothing to
> do with the "style".

Yeah, and I think everyone who's chimed in here agrees on the
substance.  It's just that @gdevenyi is leaning towards (wisely I
think), something like:

  $ git pull git://github.com/gvwilson/new-template.git

instead of landing this PR, because (assuming gvwilson/new-template is
adopted by SWC's maintainers [1](http://lists.software-carpentry.org/pipermail/maintainers_lists.software-carpentry.org/2016-May/000275.html)), it will have a clear maintenance
plan.
